### PR TITLE
Implementation of set amount parking spaces

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
The functionality of setting the parking spaces by the user was implemented, whit the MVP y el DialogFragment.

Unit Test
image
image

Video
https://user-images.githubusercontent.com/84338513/126542938-bbdbb9ba-98c6-4b8a-90db-25d492a354d8.mp4

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/model/SpacesParkingDialogModel.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class SpacesParkingDialogModel : SpacesParkingDialogContract.DialogModel {
+
+    companion object {
+        private const val SPACE_DEFAULT = 0
+    }
+
+    private var spaces: Int = SPACE_DEFAULT
+
+    override fun checkSpacesInput(spaces: String) {
+        this.spaces = if (spaces.isEmpty()) SPACE_DEFAULT else spaces.toInt()
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystemkotlin/mvp/model/SpacesParkingDialogModel.kt]: remember to use the curly bracket { }

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/model/SpacesParkingDialogModel.kt (hunk 1 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/model/SpacesParkingDialogModel.kt]: done

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/parkingsystemkotlin/fragment/SpacesParkingDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        }
+    }
+
+    private lateinit var binding: FragmentSpacesParkingDialogBinding
+    private lateinit var presenter: SpacesParkingDialogContract.DialogPresenter
+    private lateinit var listenerDialogFragment: ListenerDialogFragment
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        binding = FragmentSpacesParkingDialogBinding.inflate(inflater, container, false)
+        return binding.root
```
Comment (lucianoeli) [inline app/src/main/java/com/example/parkingsystemkotlin/fragment/SpacesParkingDialogFragment.kt:23]: you can use only one line 
..): View = FragmentSpacesParkingDialogBinding.inflate(inflater, container, false).root

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/ParkingView.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    override fun showDialog(listenerDialogFragment: ListenerDialogFragment) {
+        getFragmentManager()?.let {
+            val dialogFragment = SpacesParkingDialogFragment.newInstance(listenerDialogFragment)
+            dialogFragment.show(it, TAG)
+        }
+    }
+
+    companion object {
+        private const val TAG: String = "FRAGMENT"
```
Comment (lucianoeli) [inline app/src/main/java/com/example/parkingsystemkotlin/mvp/view/ParkingView.kt]: more descriptive tag value please.
this is useful to detect in logcat wich fragment is shown. So try to identify It on the tag
 `private const val TAG: String = "SPACES_PARKING_FRAGMENT"` or something like that

Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/view/ParkingView.kt]: done

File: app/src/main/java/com/example/parkingsystemkotlin/fragment/SpacesParkingDialogFragment.kt (hunk 2 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystemkotlin/fragment/SpacesParkingDialogFragment.kt:23]: done

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/model/ParkingModel.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    private var spaces: Int = SPACE_DEFAULT
+
+    override fun setSpaces(spaces: Int) {
+        this.spaces = spaces
+    }
+
+    override fun getSpaces(): Int = spaces
+
+    companion object {
+        private const val SPACE_DEFAULT = 0
```
Comment (nicolbidone) [inline app/src/main/java/com/example/parkingsystemkotlin/mvp/model/ParkingModel.kt]: This should be in a constant utils file, you have the same const in more than one file

Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/model/ParkingModel.kt]: Done

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/ActivityView.kt [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.FragmentManager
+import java.lang.ref.WeakReference
+
+open class ActivityView(activity: Activity) {
+
+    private val activityRef: WeakReference<Activity> = WeakReference(activity)
+
+    val activity: Activity?
+        get() = activityRef.get()
```
[the remark this answers is not in the document]
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/ActivityView.kt:14]: got it

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/FragmentView.kt [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import android.content.Context
+import androidx.fragment.app.Fragment
+import java.lang.ref.WeakReference
+
+open class FragmentView(fragment: Fragment) {
+
+    private val fragmentRef: WeakReference<Fragment> = WeakReference(fragment)
+
+    val fragment: Fragment?
+        get() = fragmentRef.get()
```
[the remark this answers is not in the document]
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/FragmentView.kt:12]: got it

**Review by nicolbidone — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**